<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Chatbot</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QSplitter" name="centralSplitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="handleWidth">
       <number>10</number>
      </property>
      <widget class="QTabWidget" name="mainTabWidget">
       <property name="currentIndex">
        <number>6</number>
       </property>
       <widget class="QWidget" name="welcomeTab">
        <attribute name="title">
         <string>Inicio</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_7">
         <item row="0" column="0">
          <widget class="QStackedWidget" name="initStackWidget">
           <property name="currentIndex">
            <number>0</number>
           </property>
           <widget class="QWidget" name="welcomePage">
            <layout class="QGridLayout" name="gridLayout_6">
             <item row="0" column="0">
              <layout class="QGridLayout" name="welcomePageLayout">
               <item row="1" column="0" colspan="5">
                <widget class="QLabel" name="welcomeLabel">
                 <property name="text">
                  <string>What do you want to do?</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignCenter</set>
                 </property>
                </widget>
               </item>
               <item row="4" column="0" colspan="5">
                <spacer name="verticalSpacer_4">
                 <property name="orientation">
                  <enum>Qt::Vertical</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>40</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item row="2" column="0" colspan="5">
                <spacer name="verticalSpacer_5">
                 <property name="orientation">
                  <enum>Qt::Vertical</enum>
                 </property>
                 <property name="sizeType">
                  <enum>QSizePolicy::Fixed</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item row="3" column="1">
                <widget class="QToolButton" name="createChatbotButton">
                 <property name="text">
                  <string>Create new chatbot</string>
                 </property>
                 <property name="icon">
                  <iconset resource="../res/chatbot_theme.qrc">
                   <normaloff>:/icons/new_chatbot.png</normaloff>:/icons/new_chatbot.png</iconset>
                 </property>
                 <property name="iconSize">
                  <size>
                   <width>48</width>
                   <height>48</height>
                  </size>
                 </property>
                 <property name="toolButtonStyle">
                  <enum>Qt::ToolButtonTextUnderIcon</enum>
                 </property>
                 <property name="autoRaise">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="3" column="3">
                <widget class="QToolButton" name="openChatbotButton">
                 <property name="text">
                  <string>Open existent chatbot</string>
                 </property>
                 <property name="icon">
                  <iconset resource="../res/chatbot_theme.qrc">
                   <normaloff>:/icons/open_chatbot.png</normaloff>:/icons/open_chatbot.png</iconset>
                 </property>
                 <property name="iconSize">
                  <size>
                   <width>48</width>
                   <height>48</height>
                  </size>
                 </property>
                 <property name="toolButtonStyle">
                  <enum>Qt::ToolButtonTextUnderIcon</enum>
                 </property>
                 <property name="autoRaise">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="0" column="0" colspan="5">
                <spacer name="verticalSpacer_6">
                 <property name="orientation">
                  <enum>Qt::Vertical</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>40</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item row="5" column="0" colspan="5">
                <spacer name="verticalSpacer_7">
                 <property name="orientation">
                  <enum>Qt::Vertical</enum>
                 </property>
                 <property name="sizeType">
                  <enum>QSizePolicy::Fixed</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>40</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item row="3" column="2">
                <widget class="QToolButton" name="openLastChatbotButton">
                 <property name="text">
                  <string>Open my last chatbot</string>
                 </property>
                 <property name="icon">
                  <iconset resource="../res/chatbot_theme.qrc">
                   <normaloff>:/icons/open_last_chatbot.png</normaloff>:/icons/open_last_chatbot.png</iconset>
                 </property>
                 <property name="iconSize">
                  <size>
                   <width>48</width>
                   <height>48</height>
                  </size>
                 </property>
                 <property name="toolButtonStyle">
                  <enum>Qt::ToolButtonTextUnderIcon</enum>
                 </property>
                 <property name="autoRaise">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="3" column="4">
                <spacer name="horizontalSpacer_4">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item row="3" column="0">
                <spacer name="horizontalSpacer_5">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="teachTab">
        <attribute name="title">
         <string>Teach</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout">
         <item row="1" column="0">
          <widget class="QSplitter" name="teachTabsplitter">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="frameShape">
            <enum>QFrame::NoFrame</enum>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="handleWidth">
            <number>10</number>
           </property>
           <property name="childrenCollapsible">
            <bool>false</bool>
           </property>
           <widget class="QWidget" name="verticalLayoutWidget">
            <layout class="QVBoxLayout" name="categoryTreeLayout">
             <item>
              <widget class="QTreeView" name="categoriesTree">
               <property name="mouseTracking">
                <bool>true</bool>
               </property>
               <property name="dragEnabled">
                <bool>true</bool>
               </property>
               <property name="dragDropMode">
                <enum>QAbstractItemView::InternalMove</enum>
               </property>
               <property name="defaultDropAction">
                <enum>Qt::MoveAction</enum>
               </property>
               <attribute name="headerVisible">
                <bool>false</bool>
               </attribute>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="Lvk::FE::RuleEditWidget" name="ruleEditWidget" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </widget>
         </item>
         <item row="0" column="0">
          <layout class="QHBoxLayout" name="rulesToolbarLayout">
           <item>
            <widget class="QToolButton" name="addCategoryButton">
             <property name="font">
              <font>
               <pointsize>8</pointsize>
              </font>
             </property>
             <property name="toolTip">
              <string>Add category</string>
             </property>
             <property name="icon">
              <iconset resource="../res/chatbot_theme.qrc">
               <normaloff>:/icons/add_category.png</normaloff>:/icons/add_category.png</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>24</width>
               <height>24</height>
              </size>
             </property>
             <property name="autoRaise">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="addRuleButton">
             <property name="font">
              <font>
               <pointsize>8</pointsize>
              </font>
             </property>
             <property name="toolTip">
              <string>Add rule</string>
             </property>
             <property name="icon">
              <iconset resource="../res/chatbot_theme.qrc">
               <normaloff>:/icons/add_rule.png</normaloff>:/icons/add_rule.png</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>24</width>
               <height>24</height>
              </size>
             </property>
             <property name="popupMode">
              <enum>QToolButton::MenuButtonPopup</enum>
             </property>
             <property name="autoRaise">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="rmItemButton">
             <property name="font">
              <font>
               <pointsize>8</pointsize>
              </font>
             </property>
             <property name="toolTip">
              <string>Remove rule or category</string>
             </property>
             <property name="icon">
              <iconset resource="../res/chatbot_theme.qrc">
               <normaloff>:/icons/remove.png</normaloff>:/icons/remove.png</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>24</width>
               <height>24</height>
              </size>
             </property>
             <property name="autoRaise">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_8">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="testTab">
        <attribute name="title">
         <string>Test your chatbot</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_10">
         <item row="0" column="0">
          <layout class="QHBoxLayout" name="testToolbarLayout">
           <item>
            <widget class="QToolButton" name="clearTestConvButton">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="toolTip">
              <string>Clean test conversation</string>
             </property>
             <property name="text">
              <string>Clear</string>
             </property>
             <property name="icon">
              <iconset resource="../res/chatbot_theme.qrc">
               <normaloff>:/icons/clear_conv.png</normaloff>:/icons/clear_conv.png</iconset>
             </property>
             <property name="iconSize">
              <size>
               <width>24</width>
               <height>24</height>
              </size>
             </property>
             <property name="autoRaise">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="testToolbarSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item row="1" column="0">
          <widget class="Lvk::FE::TestConversationWidget" name="testConversationText">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="overwriteMode">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="testLabel">
           <property name="text">
            <string>Test:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="Lvk::FE::TestInputText" name="testInputText"/>
         </item>
         <item row="0" column="1" rowspan="4">
          <widget class="QFrame" name="testRightFrame">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <layout class="QGridLayout" name="gridLayout_12">
            <item row="2" column="0">
             <widget class="QGroupBox" name="ruleViewGroupBox">
              <property name="title">
               <string>Rule used to reply:</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_11">
               <item row="1" column="0">
                <widget class="Lvk::FE::RuleTextView" name="ruleView">
                 <property name="readOnly">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QToolButton" name="showRuleDefButton">
                 <property name="text">
                  <string>Show definition</string>
                 </property>
                 <property name="icon">
                  <iconset resource="../res/chatbot_theme.qrc">
                   <normaloff>:/icons/show_rule.png</normaloff>:/icons/show_rule.png</iconset>
                 </property>
                 <property name="iconSize">
                  <size>
                   <width>24</width>
                   <height>24</height>
                  </size>
                 </property>
                 <property name="toolButtonStyle">
                  <enum>Qt::ToolButtonTextBesideIcon</enum>
                 </property>
                 <property name="autoRaise">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item row="4" column="0">
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="1" column="0">
             <widget class="QGroupBox" name="transfViewGroupBox">
              <property name="title">
               <string>Transformations applied:</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_13">
               <item row="0" column="0">
                <widget class="QTextEdit" name="transfView"/>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="conversationsTab">
        <attribute name="title">
         <string>Conversations</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_5">
         <item row="0" column="0">
          <widget class="Lvk::FE::ChatHistoryWidget" name="chatHistory" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="scoreTab">
        <attribute name="title">
         <string>Score</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_15">
         <property name="leftMargin">
          <number>15</number>
         </property>
         <property name="topMargin">
          <number>15</number>
         </property>
         <property name="rightMargin">
          <number>15</number>
         </property>
         <item row="0" column="0">
          <widget class="QGroupBox" name="curScoreGroupBox">
           <property name="title">
            <string>Current Score</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <layout class="QHBoxLayout" name="scoreTimeLayout">
              <property name="leftMargin">
               <number>9</number>
              </property>
              <property name="topMargin">
               <number>9</number>
              </property>
              <property name="rightMargin">
               <number>9</number>
              </property>
              <item>
               <widget class="QLabel" name="remainingTimeLabel">
                <property name="text">
                 <string notr="true">Remaining Time: 03:40:30</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="Lvk::FE::ScoreWidget" name="curScoreWidget" native="true"/>
            </item>
           </layout>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QGroupBox" name="bestScoreGroupBox">
           <property name="title">
            <string>Best Score</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="Lvk::FE::ScoreWidget" name="bestScoreWidget" native="true"/>
            </item>
           </layout>
          </widget>
         </item>
         <item row="2" column="0">
          <spacer name="scoresSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="connectTab">
        <attribute name="icon">
         <iconset resource="../res/chatbot_theme.qrc">
          <normaloff>:/icons/status_disconnected.png</normaloff>:/icons/status_disconnected.png</iconset>
        </attribute>
        <attribute name="title">
         <string>Connection</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_8">
         <item row="0" column="0">
          <widget class="Lvk::FE::ConnectionWidget" name="connectionWidget" native="true"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="verificationTab">
        <attribute name="title">
         <string>Account verification</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="Lvk::FE::ConnectionWidget" name="verificationWidget" native="true"/>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuChatbot">
    <property name="title">
     <string>Chatbot</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionImport"/>
    <addaction name="actionExport"/>
    <addaction name="separator"/>
    <addaction name="actionOptions"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuChatbot"/>
  </widget>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action">
   <property name="text">
    <string>-</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About Chatbot...</string>
   </property>
   <property name="toolTip">
    <string>About Chatbot...</string>
   </property>
  </action>
  <action name="action_2">
   <property name="text">
    <string>-</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>Save as...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionImport">
   <property name="text">
    <string>Import rules...</string>
   </property>
  </action>
  <action name="actionExport">
   <property name="text">
    <string>Export rules...</string>
   </property>
  </action>
  <action name="actionAddEmptyRule">
   <property name="text">
    <string>Add empty rule</string>
   </property>
  </action>
  <action name="actionAddCondRule">
   <property name="text">
    <string>Add rule with conditional</string>
   </property>
  </action>
  <action name="actionAddVarRule">
   <property name="text">
    <string>Add rule with variable</string>
   </property>
  </action>
  <action name="actionOptions">
   <property name="text">
    <string>Options...</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Lvk::FE::RuleTextView</class>
   <extends>QTextEdit</extends>
   <header>front-end/ruletextview.h</header>
  </customwidget>
  <customwidget>
   <class>Lvk::FE::TestConversationWidget</class>
   <extends>QTextEdit</extends>
   <header>front-end/testconversationwidget.h</header>
  </customwidget>
  <customwidget>
   <class>Lvk::FE::ChatHistoryWidget</class>
   <extends>QWidget</extends>
   <header>front-end/chathistorywidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Lvk::FE::ScoreWidget</class>
   <extends>QWidget</extends>
   <header>front-end/scorewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Lvk::FE::TestInputText</class>
   <extends>QLineEdit</extends>
   <header>front-end/testinputtext.h</header>
  </customwidget>
  <customwidget>
   <class>Lvk::FE::RuleEditWidget</class>
   <extends>QWidget</extends>
   <header>front-end/ruleeditwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Lvk::FE::ConnectionWidget</class>
   <extends>QWidget</extends>
   <header>front-end/connectionwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../res/chatbot_theme.qrc"/>
 </resources>
 <connections/>
</ui>
